requirements.txt

```diff
@@ -30,14 +30,13 @@ rouge-score>=0.1.2,<0.2.0
 sentencepiece>=0.2.0,<0.3.0
 tokenizers>=0.15.2,<0.16.0
 toml>=0.10.2,<0.11.0
-# Habana installer has 2.2.0a0+git8964477 with oneMKL
 torch>=2.2.0a0,<3.0.0 ; python_version == '3.10'
-torch>=2.2.1,<3.0.0 ; python_version != '3.10'
+torch>=2.3.0,<3.0.0 ; python_version != '3.10'
 tqdm>=4.66.2,<5.0.0
-transformers>=4.30.0,<=4.38.2
+transformers==4.38.1
```

Comment:
this is another mps specific. 4.38.1 is needed.

I can test again without it to show you the error.

Reply:
I don't like the idea of this but do we need to think about conditional imports based on environemnts?
or have separate requirement.txt files for separate HW dependencies

Reply:
Agreed with @jaideepr97 - don't see any other path forward here

Reply:
With the current limitations of Python's package metadata, there is only one way forward:
- leave the version pins in `requirements.txt` as relaxed as possible
- introduce an optional requirement `mps` with more strict requirements
- have users install MPS support with `instructlab[mps]` or `git+https://github.com/instructlab/instructlab.git@stable[mps]`

Reply:
@tiran how do I specify the dependencies in requirements.txt so that one can do`@stable[mps]`? Also, would I keep the current transformers version as well?